sippy-ng/src/releases/PayloadStreamTestFailures.js

```diff
@@ -0,0 +1,296 @@
+            pl.push(
+              <Link key={key} to={`/release/${params.release}/tags/${key}`}>
+                {key + '  '}
+              </Link>
+            )
+          }
+          return pl
+        }
+
+        return (
+          <Tooltip
+            title={`Failed in ${Object.keys(params.value).length} payloads`}
+          >
+            <Box className="clamped">{renderPayloadLinks()}</Box>
+          </Tooltip>
+        )
+      },
+    },
+  ]
+
+  const [release = props.release, setRelease] = useQueryParam(
+    'release',
+    StringParam
+  )
+  const [arch = props.arch, setArch] = useQueryParam('arch', StringParam)
+  const [stream = props.stream, setStream] = useQueryParam(
+    'stream',
+    StringParam
+  )
+
+  const [fetchError, setFetchError] = React.useState('')
+  const [isLoaded, setLoaded] = React.useState(false)
+  const [rows, setRows] = React.useState([])
+
+  const [filterModel = props.filterModel, setFilterModel] = useQueryParam(
+    'filters',
+    SafeJSONParam
+  )
+
+  const [sortField = props.sortField, setSortField] = useQueryParam(
+    'sortField',
+    StringParam
+  )
+  const [sort = props.sort, setSort] = useQueryParam('sort', StringParam)
+
+  const requestSearch = (searchValue) => {
+    const currentFilters = filterModel
+    currentFilters.items = currentFilters.items.filter(
+      (f) => f.columnField !== 'name'
+    )
+    currentFilters.items.push({
+      id: 99,
+      columnField: 'name',
+      operatorValue: 'contains',
+      value: searchValue,
+    })
+    setFilterModel(currentFilters)
+  }
+
+  const addFilters = (filter) => {
+    const currentFilters = filterModel.items.filter((item) => item.value !== '')
+
+    filter.forEach((item) => {
+      if (item.value && item.value !== '') {
+        currentFilters.push(item)
+      }
+    })
+    setFilterModel({
+      items: currentFilters,
+      linkOperator: filterModel.linkOperator || 'and',
+    })
+  }
+
+  const updateSortModel = (model) => {
+    if (model.length === 0) {
+      return
+    }
+
+    if (sort !== model[0].sort) {
+      setSort(model[0].sort)
+    }
+
+    if (sortField !== model[0].field) {
+      setSortField(model[0].field)
+    }
+  }
+
+  const fetchData = () => {
+    let queryString = ''
+    if (filterModel && filterModel.items.length > 0) {
+      queryString +=
+        '&filter=' + safeEncodeURIComponent(JSON.stringify(filterModel))
+    }
+
+    Promise.all([
+      fetch(
+        `${process.env.REACT_APP_API_URL}/api/releases/test_failures?release=${release}&arch=${arch}&stream=${stream}` +
+          queryString
+      ),
+    ])
+      .then(([analysis]) => {
+        if (analysis.status !== 200) {
+          throw new Error('server returned ' + analysis.status)
+        }
+
+        return Promise.all([analysis.json()])
+      })
+      .then(([analysis]) => {
+        if (analysis.length === 0) {
+          return <Typography variant="h5">Analysis not found.</Typography>
```

Comment:
You can't return like this in a promise, I'd remove this and put it below line 207

Reply:
Example of what I mean in https://github.com/openshift/sippy/commit/64d2caafd6ea609143eb5e2ad0b5713f245da306